### PR TITLE
refactor: topic package

### PR DESCRIPTION
**Pull request by colincclark**
- DRY up topic header, code and stories
- add loading stories
- add consistency with author profile package
- rename files to follow conventions

NB: this paves the way for AuthorHead to be done in the same way, and to be co-located in AuthorProfilew where it belongs

### CODE REVIEW

**Comment by times-tools** [comment]
If you use Expo, view our components by scanning this qr code: <br>  <br> This has been made possible through [Fructose](https://github.com/newsuk/fructose)

**Comment by aronshamash** [comment]
Looks fine but would like to not merge this until we have got the topic with article list PR merged in.

**Comment by nehasri89** [comment]
LGTM 👍

**Comment by times-tools** [comment]
If you use Expo, view our components by scanning this qr code: <br>  <br> This has been made possible through [Fructose](https://github.com/newsuk/fructose)

**Comment by times-tools** [comment]
If you use Expo, view our components by scanning this qr code: <br>  <br> This has been made possible through [Fructose](https://github.com/newsuk/fructose)

File: packages/topic/__tests__/topic-functional.js [hunk 1]
```diff
@@ -22,32 +28,39 @@ export default () => {
...
-          withImages: true
-        })}
-      >
-        <Topic {...topicProps} page={1} pageSize={pageSize} />
+      <MockedProvider mocks={mockArticles}>
+        <Topic {...props} page={1} pageSize={pageSize} />
       </MockedProvider>
     );
 
+    await delay(1500);
```
Comment (jimhunty) [inline packages/topic/__tests__/topic-functional.js]: Assume this is used to skip a loading time? You can pass in a delay parameter into the fixture generator so we should be in a position where we dont need this?

Comment (colincclark) [reply packages/topic/__tests__/topic-functional.js]: ok cool, will add this in on next one when I get the chance

**Pull request merged**